api/src/test/java/org/openmrs/api/ProgramWorkflowServiceTest.java

```diff
@@ -430,6 +435,178 @@ public void saveProgram_shouldCreateProgramWorkflows() {
 		}
 		TestUtil.assertCollectionContentsEquals(Arrays.asList("SINGLE", "MARRIED"), names);
 	}
+
+	/**
+	 * Tests if saveProgram throws API Exception with program, workflows and states having null concept 
+	 *
+	 * @see ProgramWorkflowService#saveProgram(Program)
+	 */
+	@Test
+	public void saveProgram_shouldThrowAPIExceptionGivenNullConcept() {
+		Program program = new Program();
+		program.setName("TEST PROGRAM");
+		program.setDescription("TEST PROGRAM DESCRIPTION");
+		Assertions.assertThrows(APIException.class, ()->{
+			pwsi.saveProgram(program);
```

Comment:
well we can actaully access the methods implemented using the service class ie by using pws  at this line,this will also help to remove the unwanted changes where you used pwsi

Reply:
yeah, makes sense. I'm working on the changes.

Reply:
While making changes, I realised that on using pws instead of pwsi on line 450, it throws API exception without running the saveProgram function in ProgramWorkflowServiceImpl. Don't know why is this happening.